akka-http2-support/src/main/scala/akka/http/impl/engine/http2/BufferedOutletSupport.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2
+
+import akka.stream.Outlet
+import akka.stream.stage.GraphStageLogic
+import akka.stream.stage.OutHandler
+
+trait BufferedOutletSupport { logic: GraphStageLogic ⇒
```

Comment:
I'd mark as internal api right away (even though in impl package, we still always should mark as such)